src/Polysemy/Internal.hs

```diff
@@ -401,6 +406,55 @@ subsume = hoistSem $ \u -> hoist subsume $ case decomp u of
   Left  g -> g
 {-# INLINE subsume #-}
 
+------------------------------------------------------------------------------
+-- | Interprets an effect in terms of another identical effect, given an
+-- explicit proof that the effect exists in @r@.
+--
+-- This is useful in conjunction with 'Polysemy.Internal.Union.tryMembership'.
+--
+-- @since TODO
+subsumeUsing :: forall e r a. ElemOf r e -> Sem (e ': r) a -> Sem r a
+subsumeUsing pr = hoistSem $ \u -> hoist (subsumeUsing pr) $ case decomp u of
+  Right w -> Union pr w
+  Left  g -> g
+{-# INLINE subsumeUsing #-}
+
+------------------------------------------------------------------------------
+-- | Moves all uses of an effect @e@ within the argument computation
+-- to a new @e@ placed on top of the effect stack. Note that this does not
+-- consume the inner @e@.
+--
+-- This can be used to replicate 'intercept'\/'interceptH':
+--
+-- @
+-- interceptH k = interpretH k . expose
+-- @
+--
+-- Typically, there's little reason to use 'expose' instead of
+-- 'intercept'\/'interceptH'. It's only offered for symmetry:
+-- 'subsume'\/'subsumeUsing' and 'expose'\/'exposeUsing'.
+expose :: Member e r => Sem r a -> Sem (e ': r) a
+expose = exposeUsing membership
+{-# INLINE expose #-}
+
+------------------------------------------------------------------------------
+-- | Given an explicit proof that @e@ exists in @r@, moves all uses of e@
+-- within the argument computation to a new @e@ placed on top of the effect
+-- stack. Note that this does not consume the inner @e@.
+--
+-- This is useful in conjunction with 'Polysemy.Internal.Union.tryMembership'
+-- and 'interpret'\/'interpretH' in order to conditionally perform
+-- 'intercept'-like operations.
+--
+-- @since TODO
+exposeUsing :: forall e r a. ElemOf r e -> Sem r a -> Sem (e ': r) a
+exposeUsing pr = hoistSem $ \(Union pr' wav) -> hoist (exposeUsing pr) $
+  case testEquality pr pr' of
+    Just Refl -> Union Here wav
+    _         -> Union (In pr') wav
+{-# INLINE exposeUsing #-}
```

Comment:
New operators for conditional `intercept`s. Figured this route would be more attractive than adding `interceptUsing` and `interceptHUsing`, but I'd like to hear your opinions on this.

Reply:
I'm not sure. My strategy is to add easy-to-use combinators for common operations, and more general/less-easy combinators for everything else.

Reply:
This is the biggest unresolved question. I'm actually starting to gravitate towards `interceptUsing` and `interceptHUsing`; the only issue is that it doesn't scale well if we later want to add other `intercept`-like operations which `expose` can do.

Reply:
Let's worry about scaling when we come to it. I'm happy with this! Feel free to merge when it's in a state you're happy with.

Reply:
Decided to scrap `expose/Using` and go with `interceptUsing`/`interceptUsingH`. If we eventually do need the extra power of `expose/Using`, we'll add them then.